### PR TITLE
rpc: implement submitblock

### LINKED ISSUE #344: RPC: implement submitblock

### PR DESCRIPTION
**Pull request by AnnaShaleva**
closes #344

### CODE REVIEW

File: pkg/core/blockchain.go [hunk 1]
```diff
@@ -52,6 +52,12 @@ var (
 	// ErrPolicy is returned on attempt to add transaction that doesn't
 	// comply with node's configured policy into the mempool.
 	ErrPolicy = errors.New("not allowed by policy")
+	// ErrValidationFailed is returned when trying to add block or transaction
+	// which doesn't satisfy validation rules.
+	ErrValidationFailed error
+	// ErrInvalidBlockIndex is returned when trying to add block with index
+	// other than expected height of the blockchain.
+	ErrInvalidBlockIndex error
```
Comment (roman-khimov) [inline pkg/core/blockchain.go]: Please, don't do that.

File: pkg/rpc/response/errors.go [hunk 2]
```diff
@@ -20,6 +20,18 @@ type (
 var (
 	// ErrInvalidParams represents a generic 'invalid parameters' error.
 	ErrInvalidParams = NewInvalidParamsError("", nil)
+	// ErrAlreadyExists represents SubmitBlock error with code -501
+	ErrAlreadyExists = NewSubmitBlockError(-501, "Block or transaction already exists and cannot be sent repeatedly.")
```
Comment (roman-khimov) [inline pkg/rpc/response/errors.go]: It's not just `SubmitBlock`, `sendrawtransaction` uses the same error answers.

Comment (AnnaShaleva) [reply pkg/rpc/response/errors.go]: Changed `SubmitBlockError` to `SubmitError` and added these errors to sendrawtransaction

File: pkg/core/helper_test.go [hunk 3]
```diff
@@ -391,6 +391,14 @@ func TestCreateBasicChain(t *testing.T) {
...
 		}
 	}
+
+	// Block for `submitblock` test. If you are planning to modify test chain from `testblocks.acc`,
+	// please, update param value of `submitblock positive` test.
+	b = bc.newBlock(newMinerTX())
+	buf := io.NewBufBinWriter()
+	b.EncodeBinary(buf.BinWriter)
+	require.NoError(t, buf.Err)
+	fmt.Println(hex.EncodeToString(buf.Bytes()))
```
Comment (roman-khimov) [inline pkg/core/helper_test.go]: `t.Log` would be more appropriate here.

File: pkg/rpc/server/server.go [hunk 4]
```diff
@@ -854,28 +854,24 @@ func (s *Server) sendrawtransaction(reqParams request.Params) (interface{}, erro
...
-			case network.RelayUnableToVerify:
-				err = errors.New("the block cannot be validated")
-			case network.RelayInvalid:
-				err = errors.New("block or transaction validation failed")
-			case network.RelayPolicyFail:
-				err = errors.New("one of the Policy filters failed")
-			default:
-				err = errors.New("unknown error")
-			}
+			return nil, errors.Wrap(r.Err, "transaction DecodeBinary failed")
```
Comment (roman-khimov) [inline pkg/rpc/server/server.go]: Isn't it a `response.ErrInvalidParams`?

File: pkg/core/helper_test.go (hunk 3 above)
Comment (AnnaShaleva) [reply pkg/core/helper_test.go]: Done. Also added one more test block without txs

**Pull request merged**